tex2wiki/src/tex2wiki.py

```diff
@@ -0,0 +1,413 @@
+        if ch == "$" and count % 2 == 0:
+            result += "<math>{\\displaystyle "
+            count += 1
+        elif ch == "$":
+            result += "}</math>"
+            count += 1
+        else:
+            result += ch
+
+    return result
+
+
+def format_formula(formula):
+    if formula[0] == ":":
+        return formula[1:].zfill(2)
+
+    formula = multi_split(formula.split("Formula:", 1)[1], [".", ":"])
+
+    for j in [-1, -2, -3]:
+        formula[j] = formula[j].zfill(2)
+
+    if len(formula) == 3:
+        formula = formula[0] + "." + formula[1] + ":" + formula[2]
+    else:
+        formula = ":".join(formula[:-3]) + ":" + formula[-3] + "." + formula[-2] + ":" + formula[-1]
+
+    return formula
+
+
+def format_metadata(string):
+    if string == "":
+        return ""
+
+    # modified convert_dollar_sign algorithm
+    dollar_count = 0
+    result = ""
+    for i, ch in enumerate(string):
+        if ch == "$":
+            if dollar_count % 2 == 1:
+                result += "}</math>"
+            elif dollar_count > 1:
+                result = result[:-1] + "<br />\n<math>{\\displaystyle "
+            else:
+                result += "<math>{\\displaystyle "
+            dollar_count += 1
+        else:
+            result += ch
+
+    return result.strip()
+
+
+def extract_data(data):
+    result = list()
+    for section_data in data:
+        equations = section_data[1][:-1]
+        for i, equation in enumerate(equations):
+            # formula stuff
+            try:
+                formula = format_formula(get_data_str(equation, latex="\\formula"))
+            except IndexError:  # there is no formula.
+                break
+
+            equation = equation.split("\n")[1:]
+
+            # get metadata
+            percent_list = list()
+            raw_metadata = ""
+
+            j = 0
+            while j < len(equation):
+                if "%" in equation[j]:
+                    raw_metadata += equation.pop(j)[1:].strip().strip("\n") + "\n"
+                else:
+                    j += 1
+
+            metadata = dict()
+            for data_type in METADATA_TYPES:
+                metadata[data_type] = format_metadata(get_data_str(raw_metadata, latex="\\"+data_type))
+
+            # print metadata
+
+            equations[i] = LatexEquation(formula, '\n'.join(equation), metadata)
+
+        result.append([section_data[0], equations])
+
+    return result
+
+
+def find_end(text, left_delimiter, right_delimiter, start=0):
+    net = 0  # left delimiters encountered - right delimiters encountered
+    for i, ch in enumerate(text[start:]):
+        if ch == left_delimiter:
+            net += 1
+        elif ch == right_delimiter:
+            if net == 1:
+                return i + start
+            else:
+                net -= 1
+
+    return -1
+
+
+def get_data_str(text, latex=""):
+    if latex + "{" not in text:
+        return ""
+
+    start = text.find(latex + "{")
```

Comment:
To avoid one more lookup in the string, you can get rid of the check above for not in string and just say if `start == -1 return ""`